package/com.unity.formats.usd/Runtime/Scripts/IO/Materials/MaterialImporter.cs

```diff
@@ -236,7 +236,17 @@ System.Collections.Generic.List<string> primvars
                         if (attr != null)
                         {
                             var value = attr.Get(scene.Time);
-                            uvPrimvar = pxr.UsdCs.VtValueToTfToken(value).ToString();
+
+                            // This value type can be a string or a token, depending on USD version
+                            if (string.IsNullOrEmpty(value))
+                                uvPrimvar = pxr.UsdCs.VtValueToTfToken(value).ToString();
```

Comment:
One question here, it seems like if `value` is null or is an empty string, we set set
`uvPrimvar` to `pxr.UsdCs.VtValueToTfToken(null).ToString();` or `pxr.UsdCs.VtValueToTfToken("").ToString();`

would uvPrimvar be some sort of a default in this case, or just straight go to line 248?

Reply:
`value` is a VtValue, not a string. We currently have two cases in this code depending which USD version we're on. In one case, VtValue contains string data, so can be cast to a string and String.IsNullOrEmpty will be false -> so treat as a string. In the other case, where it's a token, it can't be cast to a string so String.IsNullOrEmpty returns true and we can use VtValueToTfToken to extract the token from it.

Reply:
oh cool thanks I didnt know String.IsNullOrEmpty would return true for objects that cant be casted to a string type
And double thanks for the refactor!